src/hooks/useSearch.ts

```diff
@@ -43,3 +45,19 @@ export function useSearch() {
     addMostUsed,
   };
 }
+
+function processTeamNameQueryPrefix(
+    query: string,
+    teams: TeamInfo[]
+): { sanitizedQuery: string; relevantTeams: TeamInfo[] } {
```

Comment:
Maybe split the part for sanitizing aka splitting the query and the actual searching?